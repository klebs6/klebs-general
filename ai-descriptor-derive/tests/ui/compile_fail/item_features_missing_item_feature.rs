use ai_descriptor_derive::*;
use ai_descriptor_trait::*;

#[derive(ItemWithFeatures)]
#[ai("Potion")]
struct Potion {
    effect: PotionEffect, // PotionEffect does not implement ItemFeature
}

struct PotionEffect;

fn main() {}
