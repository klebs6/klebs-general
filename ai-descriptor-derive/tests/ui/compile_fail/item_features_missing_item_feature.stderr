error[E0599]: no method named `text` found for struct `PotionEffect` in the current scope
  --> tests/ui/compile_fail/item_features_missing_item_feature.rs:4:10
   |
4  | #[derive(ItemWithFeatures)]
   |          ^^^^^^^^^^^^^^^^ method not found in `PotionEffect`
...
10 | struct PotionEffect;
   | ------------------- method `text` not found for this struct
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `text`, perhaps you need to implement it:
           candidate #1: `ai_descriptor_trait::ItemFeature`
   = note: this error originates in the derive macro `ItemWithFeatures` (in Nightly builds, run with -Z macro-backtrace for more info)
